codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2GoAwayFrame.java

```diff
@@ -87,6 +93,17 @@ public Http2GoAwayFrame setExtraStreamIds(int extraStreamIds) {
     }
 
     @Override
+    public Http2GoAwayFrame setLastStreamId(int lastStreamId) {
+        this.lastStreamId = lastStreamId;
+        return this;
+    }
+
+    @Override
+    public Integer lastStreamId() {
+        return lastStreamId;
```

Comment:
I was thinking later it would be fairly easy to make a new Http2DifferentStream in the frame handler and convert 1:1. We're mainly really only gaining private properties on Http2Stream, but that is used in both directions multiplex codec -> frame codec and frame codec -> multiplex codec.

Although, if we're talking about having a map on each stream for properties (likely one per handler) or a map of streams for properties (likely one per handler), I guess it doesn't matter much which we go with. Having a map on each stream is more predictable, so we can use the array-based indexing, and doesn't require noticing stream death. Important properties could also become first-class fields (like maybe the child channel). But both approaches do seem a bit wonky, so I'm open to suggestions and on the fence.